### PR TITLE
feat(zigbee2mqtt): Add z2m ui link in Setup page

### LINKED ISSUE #1893: Zigbee2mqtt: Add link to Zigbee2Mqtt instance in UI
[cce66](https://community.gladysassistant.com/u/cce66)
[10h](https://community.gladysassistant.com/t/zigbee2mqtt-capteur-desactive/7156/8?u=cce66)

oui c’est cela http://adresse IP:8080/#/
[@pierre-gilles](https://community.gladysassistant.com/u/pierre-gilles)
ce serait bien en terme d’ergonomie et de convivialité d’ajouter le lien http://adresse IP:8080/#/
comme l’a fait [@Lokkye](https://community.gladysassistant.com/u/lokkye) pour l’accès Node-Red (il y a des paramètres plus complets) car on se souvient pas forcement du comment y acceder surtout pendant l’été :partying_face: :wine_glass:

image

### PR DESCRIPTION
**Pull request by cicoub13**
Fix #1893

### CODE REVIEW

**Comment by cicoub13** [comment]
CleanShot 2023-09-28 at 15 53 01

message is shown only when service is activated

**Comment by atrovato** [comment]
What will be the URL if I'm not on localhost? Using a raspberry ou a NAS hosting Gladys? Or via Gladys Plus?

**Comment by cicoub13** [comment]
> Using a raspberry ou a NAS hosting Gladys?

It will be something like http://192.168.1.11:8080 with device IP

> Or via Gladys Plus?

Link will not be showned (because not accessible from Internet)

**Comment by atrovato** [comment]
I just check your code, it will use the "localApiUrl" configuration variable to define de base URL, which means that default "localhost" will be used for most of Gladys user.
I didn't set the LAN IP of my NAS as a docker variable (LOCAL_API_URL).

**Comment by cicoub13** [comment]
I took code from @callemand about node-red https://github.com/GladysAssistant/Gladys/blob/a10436b10a62ba697dec05daf3b01ce0e884b53a/front/src/routes/integration/all/node-red/setup-page/SetupTab.jsx#L28

But let me test it in real world with a docker image 🤔

**Comment by atrovato** [comment]
Maybe you should build the entire url from the front side, according to the `window.location.hostname`, and concat with configured the port.

**Comment by cicoub13** [comment]
It's working (I don't have `LOCAL_API_URL` in Docker envs)
Port is fixed in Gladys to 8080. If you change it, you are a power user so you know where is the UI 😉 

CleanShot 2023-10-03 at 16 28 46

**Review by atrovato — APPROVED** [review]
Good for my tired eyes ;)

File: front/src/routes/integration/all/zigbee2mqtt/setup-page/actions.js [hunk 1]
```diff
@@ -38,6 +39,16 @@ const createActions = store => {
           networkModeValid: zigbee2mqttStatus.networkModeValid
         });
       }
+
+      // Compute Zigbee2Mqtt interface URL if connected locally (not through Gladys Plus Gateway)
+      let z2mUrl = null;
+      if (this.session.gatewayClient === undefined) {
```
Comment (Pierre-Gilles) [inline front/src/routes/integration/all/zigbee2mqtt/setup-page/actions.js]: I don't think it's working! this.session doesn't contain the session object, does it?

Shouldn't it be `state.session`, like we do line 27 of this file to get the state.httpClient ?

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
I'm not 100% sure the Gladys Plus part is working!

File: front/src/routes/integration/all/zigbee2mqtt/setup-page/actions.js (hunk 1 above)
Comment (cicoub13) [reply front/src/routes/integration/all/zigbee2mqtt/setup-page/actions.js]: Fixed

**Review by Pierre-Gilles — APPROVED** [review]
Thanks for the changes 🙏  Good for me !

**Pull request merged**